app/models/GMPDate.scala

```diff
@@ -39,6 +39,15 @@ case class GmpDate(day: Option[String], month: Option[String], year: Option[Stri
       false
   }
 
+  def isBefore05042046: Boolean = {
+    if (getAsLocalDate.isDefined) {
+      val thatDate = new LocalDate(2046, 4, 5)
+      getAsLocalDate.get.isBefore(thatDate)
```

Comment:
Generally tax year ends on 5 April and starts from 6 April. Not sure whether that is relevant here.